### DISCUSSION TITLE
Detecting user initiated movements for stencil elements

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2604] [Q&A] [answered]
### Introduction

I'm trying to arrange stencil element on paper and based on it's position I'm trying to embed it inside another element.
This embed must happen only when it's initiated by user (via single select or multi select of stencil elements) and not when automatic layout happens eg. grid.

The embed functionality is working, we just want to detect when element is being moved by the user. The event when embed happens is shown below:
`this.graph.on('change:position', (element: joint.dia.Element) => {}`

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
If you want to listen to a user interaction, it's easier to listen to paper events.

```ts
paper.on('element:pointermove', (elementView, evt) => {
  if (!evt.data.moved) {
    console.log(elementView, 'is start moving');
    evt.data.moved = true;
  }
});
```
